assets/src/edit-story/output/utils/getFontDeclarations.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import getGoogleFontURL from '../../utils/getGoogleFontURL';
+
+/**
+ * Returns a list of font declarations across all pages in a story.
+ *
+ * @param {Array<import('../../types').Page>} pages List of pages.
+ *
+ * @return {Array<string>} Font declarations.
+ */
+const getFontDeclarations = (pages) => {
+  const map = new Map();
+
+  for (const { elements } of pages) {
+    const textElements = elements.filter(({ type }) => type === 'text');
+
+    // Prepare font objects for later use.
+    for (const { font, fontStyle, fontWeight } of textElements) {
+      const { service, family, variants } = font;
+      if (!service || service === 'system') {
+        continue;
+      }
+
+      const serviceMap = map.get(service) || new Map();
+      map.set(service, serviceMap);
+
+      // Example: [1, 700] for bold + italic.
+      const variant = [Number(fontStyle === 'italic'), fontWeight || 400];
+
+      const fontObj = serviceMap.get(family) || { family, variants: [] };
+      const fontObjHasVariant = fontObj.variants.some(
```

Comment:
Could you pls confirm what you mean by `variants` here and ideally comment in code as well? Is the meaning of variants is more like `font-variants` CSS property? Or more of your own meaning? Or does Google Fonts send us something specific here? I'm asking because the `font-variant` CSS doesn't support italic/weight and such. So, I just wanted to make sure there's no confusion here.

Reply:
I see, this wording might be non-ideal.

Throughout the code here, variants refer to what Google Fonts calls axis tuples. A tuple is a possible style+weight combination.

Reply:
TBH, I'd really like to call them variants :) Just the confusion with CSS naming. But after some searching around, I think "variants" is really the best name for us. We should just stress it in comments here that this is what it stands for.